src/common/xo/migrate-vm-modal/index.js

```diff
@@ -137,10 +134,12 @@ export default class MigrateVmModalBody extends BaseComponent {
         const pif = pifs[pifId]
         return pif.ip && networks[vif.$network].name_label === networks[pif.$network].name_label
       })
-      defaultNetworks[vif.id] = defaultPif && networks[defaultPif.$network].id
+      let defaultNetwork = defaultPif && networks[defaultPif.$network]
+      defaultNetworks[vif.id] = defaultPif && defaultNetwork && defaultNetwork.id
       // ...or the first network in the target host networks list that has an IP.
       if (!defaultNetworks[vif.id]) {
-        defaultNetworks[vif.id] = networks[pifs[find(host.$PIFs, pif => pifs[pif].ip)].$network].id
+        defaultNetwork = networks[pifs[find(host.$PIFs, pif => pifs[pif].ip)].$network]
+        defaultNetworks[vif.id] = defaultNetwork && defaultNetwork.id
       }
```

Comment:
// Should probably be in `xo/utils.js`.
const getDefaultNetworkForVif = (vif, host, pifs, networks) => {
  const nameLabel = networks[vif.$network].name_label
  let defaultNetwork
  forEach(host.$PIFs, pifId => {
    const pif = pifs[pifId]
    if (pif.ip && networks[pif.$network].name_label === nameLabel) {
      defaultNetwork = pif.$network
      return false
    }
  })
  return defaultNetwork
}

const defaultNetwork = invoke(() => {
  // First PIF with an IP.
  const pif = pifs[find(host.$PIFs, pif => pifs[pif].ip)]

  return pif && pif.$network
})

forEach(vifs, vif => {
  defaultNetworksForVif[vif.id] = (
    getDefaultNetworkForVif(vif, host, pifs, networks) ||
    defaultNetwork
  )
})